File: backend/voting-app/src/main/java/com/tu/votingapp/entities/referendum/ReferendumOptionEntity.java
package com.tu.votingapp.entities.referendum;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "referendum_options")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class ReferendumOptionEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "option_text")
    private String optionText;

    @Column(name = "vote_count")
    private int voteCount;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "referendum_id")
    private ReferendumEntity referendum;

    public ReferendumOptionEntity(Long id) {
        this.id = id;
    }
}

File: frontend/android/app/src/main/java/com/example/android/entity/referendum/ReferendumOptionEntity.kt
package com.example.android.entity.referendum

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "referendum_options")
data class ReferendumOptionEntity(
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "id")
    val id: Long? = null,

    @ColumnInfo(name = "option_text")
    val optionText: String,

    @ColumnInfo(name = "vote_count")
    val voteCount: Int = 0,

    @ColumnInfo(name = "referendum_id")
    val referendumId: Long
)